src/teleop/gui/src/components/Cameras.vue

```diff
@@ -0,0 +1,165 @@
+        <button class="rounded button" v-on:click="addCameraName()">Change name</button>
+      </div>
+      <div class="cameraselection">
+        <CameraSelection class="cameraspace1" v-bind:camsEnabled="camsEnabled" v-bind:names="names" v-bind:capacity="parseInt(capacity)" v-on:cam_index="setCamIndex($event)"/>
+      </div>
+      <h3>All Cameras</h3>
+      Capacity: <input class="rounded" type='Number' min="2" max="4" v-model ='capacity'>
+      <div class="camerainfo" v-for="i in camsEnabled.length" :key="i">
+        <CameraInfo v-if="camsEnabled[i-1]" v-bind:name="names[i-1]" v-bind:id="i-1" v-on:newQuality="changeQuality($event)" v-on:swapStream="swapStream($event)" v-bind:stream="getStreamNum(i-1)"></CameraInfo>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  import ROSLIB from 'roslib/src/RosLib'
+  import CameraSelection from '../components/CameraSelection.vue'
+  import CameraInfo from '../components/CameraInfo.vue'
+  
+  export default {
+    data() {
+      return {
+        camsEnabled: new Array(9).fill(false),
+        names: Array.from({length: 9}, (_,i) => "Camera: " + i),
+        cameraIdx: 1,
+        cameraName: "",
+        capacity: 2,
+        qualities: new Array(9).fill(1),
+        streamOrder: [-1, -1, -1, -1]
+      }
+    },
+
+    props: {
+      numCams: {
+        type: Number,
+        required: true
+      },
+      mission: {
+        type: String,
+        required: true
+      },
+      channel: {
+        type: String,
+        required: true
+      },
+      primary: {
+        type: Boolean,
+        required: true
+      }
+
+    },
+
+
+    methods: {
+
+      setCamIndex: function (index) { //every time a button is pressed, it changes cam status and adds/removes from stream
+        this.camsEnabled.splice(index, 1, !this.camsEnabled[index]);
+        this.changeStream(index);
+        this.sendCameras();
+      },
+
+      sendCameras: function() { //sends cameras to a service to display on screen
+        var msgs = [];
+        for(var i = 0; i < 4; i++){
+          var camId = this.streamOrder[i];
+          var res = this.qualities[camId];
+          msgs.push(new ROSLIB.Message({device: camId, resolution: res})); //CameraCmd msg
+        }
+
+        var changeCamsService = new ROSLIB.Service({
+              ros : this.$ros,
+              name : 'change_cameras',
+              serviceType : 'ChangeCameras'
+          });
+
+        var request = new ROSLIB.ServiceRequest({primary: this.primary, camera_cmds: msgs});
+        changeCamsService.callService(request, (result) => {});
+
+      },
+
+      addCameraName: function() {
+        this.names.splice(this.cameraIdx, 1, this.cameraName)
+      },
+
+      changeQuality({index, value}){
+        this.qualities.splice(index,1,value);
+      },
+
+      swapStream({prev, newest}){
+        var temp = this.streamOrder[prev];
+        this.streamOrder.splice(prev,1, this.streamOrder[newest]);
```

Comment:
Also for arrays we can use Vue.set as opposed to slice, I think it is less confusing syntax wise since splice can be used for multiple different things. 
Documentation on this here: https://v2.vuejs.org/v2/guide/reactivity.html#For-Arrays